### PR TITLE
Remove opentelemetry as the default metrics library

### PR DESCRIPTION
**Pull request by emschwartz**
Also make it so that enabling multiple metrics libraries will encode metrics multiple times. This is so that we don't need to make the features mutually exclusive.

Resolves https://github.com/autometrics-dev/autometrics-rs/issues/89, https://github.com/autometrics-dev/autometrics-rs/issues/95

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,15 +17,18 @@ jobs:
...
-      - run: cargo test --no-default-features --features=prometheus-exporter,metrics --tests
-      - run: cargo test --no-default-features --features=prometheus-exporter,prometheus --tests
-      - run: cargo test --no-default-features --features=prometheus-exporter,prometheus-client --tests
+      - run: cargo test --features=prometheus-exporter,metrics
+      - run: cargo test --features=prometheus-exporter,prometheus --tests
+      - run: cargo test --features=prometheus-exporter,prometheus-client --tests
+      # The tests using opentelemetry are currently failing because of this issue:
+      # https://github.com/open-telemetry/opentelemetry-rust/issues/1070
+      # We should re-enable them once 0.20 or 0.19.1 are released
+      # - run: cargo test --features=prometheus-exporter,opentelemetry
```
Comment (mellowagain) [inline .github/workflows/ci.yml:27]: maybe we can keep it enabled as a separate build test but allow it to fail (`continue-on-error: true`) so we don't forget about it

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,15 +17,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Users must configure the metrics library they want to use autometrics with
+- Previously,
```
Comment (mellowagain) [inline CHANGELOG.md:21]: previously what

File: autometrics/README.md [hunk 3]
```diff
@@ -115,8 +115,9 @@ fn main() {
 
 #### Metrics Libraries
 
-Configure the crate that autometrics will use to produce metrics by using one of the following feature flags:
+**Required:** Configure the crate that autometrics will use to produce metrics by using one of the following feature flags:
```
Comment (mellowagain) [inline autometrics/README.md:118]: maybe we should also document what the prioritization order is when multiple are enabled

File: autometrics/src/prometheus_exporter.rs [hunk 4]
```diff
@@ -2,47 +2,89 @@ use crate::HISTOGRAM_BUCKETS;
...
+    Format(#[from] std::fmt::Error),
+}
+
+static GLOBAL_EXPORTER: Lazy<GlobalPrometheus> = Lazy::new(|| GlobalPrometheus {
+    #[cfg(feature = "metrics")]
+    metrics_exporter: PrometheusBuilder::new()
+        .set_buckets(&HISTOGRAM_BUCKETS)
+        .expect("Failed to set histogram buckets")
+        .install_recorder()
+        .expect("Failed to install recorder"),
```
Comment (mellowagain) [inline autometrics/src/prometheus_exporter.rs:31]: is it really appropriate for a library to panic? it'd be better if we just returned `Result`s so end user programs don't end up panic-ing

File: autometrics/README.md (hunk 3 above)
Comment (emschwartz) [reply autometrics/README.md:118]: I changed it so it'll use all of them if multiple are enabled

File: .github/workflows/ci.yml (hunk 1 above)
Comment (emschwartz) [reply .github/workflows/ci.yml:27]: Neat, didn't know about that

**Review by mellowagain — APPROVED** [review]

**Pull request merged**